esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5008,17 +5027,19 @@ bool CWsDfuEx::onListHistory(IEspContext &context, IEspListHistoryRequest &req,
         PROGLOG("onListHistory: %s", req.getName());
 
         MemoryBuffer xmlmap;
+        IArrayOf<IEspHistory> arrHistory;
         Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(req.getName(),userdesc.get());
         if (file)
         {
             IPropertyTree *history = file->queryHistory();
             if (history)
-                history->serialize(xmlmap);
+                storeHistoryTreeToArray(history, arrHistory);
+
+            if (arrHistory.ordinality())
+                resp.setFileHistory(arrHistory);
         }
         else
             throw MakeStringException(ECLWATCH_FILE_NOT_EXIST,"CWsDfuEx::onListHistory: Could not find file '%s'.", req.getName());
-
-        resp.setXmlmap(xmlmap);
```

Comment:
For backward compatibility, if the client is using the old interface version, don't you need to set xmlmap?  Not sure which version the original response was added, 1.35?  You still want people using that client to be able to work.